### ISSUE TITLE
Should the (surrogate) learner be part of the AcqF?

### ISSUE THREAD

**Issue by jakob-r** [mlr-org/mlr3mbo#6]

**Comment by jakob-r** [comment]
I think no. Because:
1. You want to be able to change the AcqF during optimization (You could still do this in a cumbersome way by taking the old learner and put it into the new AcqF object)
2. You want to be able to use multiple AcqF on one surrogate model for multipoint prediction

**Comment by berndbischl** [comment]
i dont understand what you are posting here. the acquisition-function surely somehow needs to access the learner in order for the function to be defined. 
it would be easier if you could post a concrete design here, with concrete examples for 1. and 2.

and currently doint what you want to do in 1. seems very easy to achive?

acqf = AcqFunctionMean$new()
acqf$set_up(param_set, task, surrogate) 
# propose a point, do something, iterate
acqf = AcqFunctionCB$new(lam = 3)
acqf$set_up(param_set, task, surrogate) 
# propose a point, do something, iterate

seems super simple?

**Comment by jakob-r** [comment]
Yes. We construct the AcqF without a Surrogate but internally it needs to have a pointer to the surrogate model (not learner) before the AcqF gets called. Where and when the AcqF is "set_up" or "prepared" can be discussed.

**Comment by berndbischl** [comment]
There is no difference in mlr3 anymore between a learner and a model?

**Comment by jakob-r** [comment]
Let me rephrase this sentence in mlr3 specific terms then: "It needs to have a pointer to the learner object that is already trained before the AcqF gets called"

**Comment by jakob-r** [comment]
it is
